File: backend-repo/FaniverseBE/src/main/java/fantastic/faniverse/chat/presentation/request/MessageRequest.java
package fantastic.faniverse.chat.presentation.request;

import fantastic.faniverse.chat.domain.ChatMessage;
import fantastic.faniverse.chat.domain.ChatRoom;
import fantastic.faniverse.chat.domain.MessageType;
import fantastic.faniverse.user.entity.User;
import lombok.Getter;

import java.time.LocalDateTime;

@Getter
public class MessageRequest {
    private Long userId;
    private String message;
    private String imageUrl;

    public ChatMessage toEntity(User sender, ChatRoom chatRoom) {
        ChatMessage.ChatMessageBuilder builder = ChatMessage.builder()
                .createdAt(LocalDateTime.now())
                .chatRoom(chatRoom)
                .senderId(sender.getId());

        if (imageUrl == null) {
            return builder.message(message)
                    .messageType(MessageType.TEXT)
                    .build();
        }
        return builder
                .imageLink(imageUrl)
                .messageType(MessageType.IMAGE)
                .build();
    }
}

File: frontend-repo/FaniverseFE2/app/src/main/java/com/example/login/model/MessageRequest.kt
package com.example.login.model

import java.util.Date

data class MessageRequest(
    val userId: Long,
    val message: String?,
    val imageUrl: String?
) {
    fun toEntity(senderId: Long, chatRoomId: Long): ChatMessage {
        val createdAt = Date() // 현재 시간을 Date 객체로 생성

        return if (imageUrl == null) {
            ChatMessage(
                senderId = senderId,
                chatRoomId = chatRoomId,
                message = message ?: "",
                imageLink = null,
                messageType = MessageType.TEXT,
                createdAt = createdAt // Date 객체를 사용
            )
        } else {
            ChatMessage(
                senderId = senderId,
                chatRoomId = chatRoomId,
                message = null,
                imageLink = imageUrl,
                messageType = MessageType.IMAGE,
                createdAt = createdAt // Date 객체를 사용
            )
        }
    }
}
